src/modules/core/link_timeremap.c

```diff
@@ -0,0 +1,417 @@
+	int s = 0;
+	uint8_t* p = *image;
+	for( s = 0; s < size; s++ )
+	{
+		int16_t sum = 0;
+		int i = 0;
+		for( i = 0; i < image_count; i++ )
+		{
+			sum += *(images[i]);
+			images[i]++;
+		}
+		*p = sum / image_count;
+		p++;
+	}
+	mlt_frame_set_image( frame, *image, size, mlt_pool_release );
+	mlt_properties_set_int( MLT_FRAME_PROPERTIES(frame), "format", *format );
+	mlt_properties_set_int( MLT_FRAME_PROPERTIES(frame), "width", *width );
+	mlt_properties_set_int( MLT_FRAME_PROPERTIES(frame), "height", *height );
+
+	return 0;
+}
+
+static int link_get_image_nearest( mlt_frame frame, uint8_t** image, mlt_image_format* format, int* width, int* height, int writable )
+{
+	mlt_link self = (mlt_link)mlt_frame_pop_get_image( frame );
+	mlt_properties unique_properties = mlt_frame_get_unique_properties( frame, MLT_LINK_SERVICE(self) );
+	if ( !unique_properties )
+	{
+		return 1;
+	}
+	double source_time = mlt_properties_get_double( unique_properties, "source_time");
+	double source_fps = mlt_properties_get_double( unique_properties, "source_fps");
+	mlt_position in_frame_pos = floor( source_time * source_fps );
+	char key[19];
+	sprintf( key, "%d", in_frame_pos );
+
+	mlt_frame src_frame = (mlt_frame)mlt_properties_get_data( unique_properties, key, NULL );
+	if ( src_frame )
+	{
+		uint8_t* in_image;
+		if ( !mlt_frame_get_image( src_frame, &in_image, format, width, height, 0 ) )
+		{
+			int size = mlt_image_format_size( *format, *width, *height, NULL );
+			*image = mlt_pool_alloc( size );
+			memcpy( *image, in_image, size );
+			mlt_frame_set_image( frame, *image, size, mlt_pool_release );
+			mlt_properties_set_int( MLT_FRAME_PROPERTIES(frame), "format", *format );
+			mlt_properties_set_int( MLT_FRAME_PROPERTIES(frame), "width", *width );
+			mlt_properties_set_int( MLT_FRAME_PROPERTIES(frame), "height", *height );
+
+			uint8_t* in_alpha = mlt_frame_get_alpha( src_frame );
+			if ( in_alpha )
+			{
+				size = *width * *height;
+				uint8_t* out_alpha = mlt_pool_alloc( size );
+				memcpy( out_alpha, in_alpha, size );
+				mlt_frame_set_alpha( frame, out_alpha, size, mlt_pool_release );
+			};
+			return 0;
+		}
+	}
+
+	return 1;
+}
+
+static int link_get_frame( mlt_link self, mlt_frame_ptr frame, int index )
+{
+	mlt_properties properties = MLT_LINK_PROPERTIES( self );
+	mlt_position position = mlt_producer_position( MLT_LINK_PRODUCER( self ) );
+	mlt_position length = mlt_producer_get_length( MLT_LINK_PRODUCER( self ) );
+	double source_time = 0.0;
+	double source_duration = 0.0;
+	double source_fps = mlt_producer_get_fps( self->next );
+
+	int result = 0;
+
+	// Create a  frame
+	*frame = mlt_frame_init( MLT_LINK_SERVICE(self) );
+	mlt_frame_set_position( *frame, mlt_producer_position( MLT_LINK_PRODUCER(self) ) );
+	mlt_properties unique_properties = mlt_frame_unique_properties( *frame, MLT_LINK_SERVICE(self) );
+
+	// Calculate the frames from the next link to be used
+	if ( !mlt_properties_exists( properties, "map" ) )
```

Comment:
If the "map" property is not set, then this link is a basic frame rate converter.